Assets/MixedRealityToolkit.SDK/Features/Utilities/Solvers/TapToPlace.cs

```diff
@@ -0,0 +1,379 @@
+        /// remain facing the camera while IsBeingPlaced is true.  If true, the object will rotate according to the surface normal
+        /// if there is a hit.
+        /// </summary>
+        public bool RotateAccordingToSurface
+        {
+            get => rotateAccordingToSurface;
+            set => rotateAccordingToSurface = value;
+        }
+
+        [SerializeField]
+        [Tooltip("Array of LayerMask to execute from highest to lowest priority. First layermask to provide a raycast hit will be used by component.")]
+        private LayerMask[] magneticSurfaces = { UnityEngine.Physics.DefaultRaycastLayers };
+
+        /// <summary>
+        /// Array of LayerMask to execute from highest to lowest priority. First layermask to provide a raycast hit will be used by component.
+        /// </summary>
+        public LayerMask[] MagneticSurfaces
+        {
+            get => magneticSurfaces;
+            set => magneticSurfaces = value;
+        }
+
+        [SerializeField]
+        [Tooltip("If true and in the Unity Editor, the normal of the raycast hit will be drawn in yellow.")]
+        private bool debugEnabled = true;
+
+        /// <summary>
+        /// If true and in the Unity Editor, the normal of the raycast hit will be drawn in yellow.
+        /// </summary>
+        public bool DebugEnabled
+        {
+            get => debugEnabled;
+            set => debugEnabled = value;
+        }
+
+        [SerializeField]
+        [Tooltip("This event is triggered once when the game object to place is selected.")]
+        private UnityEvent onPlacingStarted = new UnityEvent();
+
+        /// <summary>
+        /// This event is triggered once when the game object to place is selected.
+        /// </summary>
+        public UnityEvent OnPlacingStarted
+        {
+            get => onPlacingStarted;
+            set => onPlacingStarted = value;
+        }
+
+        [SerializeField]
+        [Tooltip("This event is triggered once when the game object to place is unselected, placed.")]
+        private UnityEvent onPlacingStopped = new UnityEvent();
+
+        /// <summary>
+        /// This event is triggered once when the game object to place is unselected, placed.
+        /// </summary>
+        public UnityEvent OnPlacingStopped
+        {
+            get => onPlacingStopped;
+            set => onPlacingStopped = value;
+        }
+
+        /// <summary>
+        /// The current game object layer before it is temporarily switched to IgnoreRaycast while placing the game object.
+        /// </summary>
+        protected internal int GameObjectLayer { get; protected set; }
+
+        protected internal bool IsColliderPresent => gameObject != null ? gameObject.GetComponent<Collider>() != null : false;
+
+        private int ignoreRaycastLayer;
+
+        // The current ray is based on the TrackedTargetType (Controller Ray, Head, Hand Joint).
```

Comment:
Might be good to convert this into a `summary` tag, since protected fields show up in the docs.

Reply:
Added in #7371